programs/zstdcli.c

```diff
@@ -1213,6 +1227,23 @@ int main(int const argCount, const char* argv[])
         if (adaptMin > cLevel) cLevel = adaptMin;
         if (adaptMax < cLevel) cLevel = adaptMax;
 
+        if (showDefaultCParams) {
+            size_t fileNb;
+            for (fileNb = 0; fileNb < (size_t)filenames->tableSize; fileNb++) {
+                unsigned long long fileSize = UTIL_getFileSize(filenames->fileNames[fileNb]);
+                const size_t dictSize = dictFileName != NULL ? (size_t)UTIL_getFileSize(dictFileName) : 0;
+                const ZSTD_compressionParameters cParams = ZSTD_getCParams(cLevel, fileSize, dictSize);
+                if (fileSize != UTIL_FILESIZE_UNKNOWN) DISPLAY("%s (%u bytes)\n", filenames->fileNames[fileNb], (unsigned)fileSize);
+                else DISPLAY("%s (src size unknown)\n", filenames->fileNames[fileNb]);
+                DISPLAY(" - windowLog    : %u\n", cParams.windowLog);
+                DISPLAY(" - chainLog     : %u\n", cParams.chainLog);
+                DISPLAY(" - searchLog    : %u\n", cParams.searchLog);
+                DISPLAY(" - minMatch     : %u\n", cParams.minMatch);
+                DISPLAY(" - targetLength : %u\n", cParams.targetLength);
+                DISPLAY(" - strategy     : %s\n", ZSTD_strategyMap[(int)cParams.strategy]);
```

Comment:
Ensure that this array read cannot result in an out-of-bound access.
An `assert()` is good enough for this case, since it's not supposed to happen.